### PR TITLE
Enable sending Key/Value Attributes to Queue

### PR DESCRIPTION
**Pull request by jeffsteinmetz**
Fix bug that prevents receiving Message Attributes.

### CODE REVIEW

File: src/main/scala/com/kifi/franz/FakeSQSQueue.scala [hunk 1]
```diff
@@ -17,7 +17,7 @@ trait FakeSQSQueue[T] extends SQSQueue[T] {
 
   override def initQueueUrl(): String = ""
 
-  override def send(msg: T): Future[MessageId] = Future.successful(MessageId(""))
+  override def send(msg: T, messageAttributes: Option[Map[String,String]]): Future[MessageId] = Future.successful(MessageId(""))
```
Comment (jeffsteinmetz) [inline src/main/scala/com/kifi/franz/FakeSQSQueue.scala]: BTW, right now just supports simple string values.

**Comment by stkem** [comment]
Why did you change code formatting and the mode of all the files? Also, what is the benefit of having message attributes over just having that same data in the payload itself?

**Comment by jeffsteinmetz** [comment]
The file mode was not an intended change.  Not sure why git merged with file mode changes.  Actually, first time I have seen that.
 
I ran across a requirement to pass attributes along with data that is posted, where it's not desired to change the actual payload itself.
Message attributes can be used by the receiver of the message to help decide how to handle the message without having to first process the message body.

**Comment by jeffsteinmetz** [comment]
just fixed file mode + merge conflict

File: src/main/scala/com/kifi/franz/SQSQueue.scala [hunk 2]
```diff
@@ -10,13 +9,14 @@ import scala.language.implicitConversions
...
-  CreateQueueRequest
+SendMessageRequest,
+GetQueueUrlRequest,
+ReceiveMessageRequest,
+DeleteMessageRequest,
+SendMessageResult,
+ReceiveMessageResult,
+CreateQueueRequest,
+MessageAttributeValue
 }
```
Comment (stkem) [inline src/main/scala/com/kifi/franz/SQSQueue.scala]: Why the format change?

File: src/main/scala/com/kifi/franz/SQSQueue.scala [hunk 3]
```diff
@@ -50,11 +50,31 @@ trait SQSQueue[T]{
     }
   }
 
+  def stringMessageAttribute( attributeValue: String ): MessageAttributeValue =
```
Comment (stkem) [inline src/main/scala/com/kifi/franz/SQSQueue.scala]: Why is this a public method?

File: src/main/scala/com/kifi/franz/SQSQueue.scala [hunk 4]
```diff
@@ -50,11 +50,31 @@ trait SQSQueue[T]{
...
+    attr
+  }
+
+  def send(msg: T ): Future[MessageId] = {
+    send (msg, None)
+  }
+
+  def send(msg: T, messageAttributes: Option[Map[String, String]] = None): Future[MessageId] = {
 
-  def send(msg: T): Future[MessageId] = {
```
Comment (stkem) [inline src/main/scala/com/kifi/franz/SQSQueue.scala]: Empty Line?

File: src/main/scala/com/kifi/franz/SQSQueue.scala [hunk 5]
```diff
@@ -50,11 +50,31 @@ trait SQSQueue[T]{
     }
   }
 
+  def stringMessageAttribute( attributeValue: String ): MessageAttributeValue =
+  {
```
Comment (stkem) [inline src/main/scala/com/kifi/franz/SQSQueue.scala]: No newline before the opening brace please.

**Comment by stkem** [comment]
Please follow code style used in the existing codebase (I've made a few comments).

**Comment by jeffsteinmetz** [comment]
Probably IntelliJ vs. Sublime text inadvertent formatting.
I think it should better match the previous style now.

File: src/main/scala/com/kifi/franz/SQSQueue.scala [hunk 6]
```diff
@@ -85,11 +106,12 @@ trait SQSQueue[T]{
...
+              attributes = rawMessage.getAttributes.asScala.toMap,
+              messageAttributes = rawMessage.getMessageAttributes.asScala.toMap)
           })
-          } catch {
-            case t: Throwable => p.failure(t)
-          }
+        } catch {
+          case t: Throwable => p.failure(t)
+        }
       }
```
Comment (jeffsteinmetz) [inline src/main/scala/com/kifi/franz/SQSQueue.scala]: The bracket indents were changed to match earlier ```try { ... ``` indent level.

Comment (stkem) [inline src/main/scala/com/kifi/franz/SQSQueue.scala]: :+1:

**Pull request merged**